### PR TITLE
Add dummy scripts to test TG execution flow

### PR DESCRIPTION
**Pull request by yasassri**
## Purpose
We need a set of dummy scripts to test the TG execution flow, this PR will add dummy scripts to deploy a tomcat server on AWS and testing it with Jmeter scripts. Fixes: #282 

## Goals
Adds Test scripts

## Approach

untitled diagram 1
282

### CODE REVIEW

File: test-scripts/tomcat-deployment/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+Dummy Scripts to Test TG execution
```
Comment (vidurananayakkara) [inline test-scripts/tomcat-deployment/README.md]: Shall we make this a title (## Dummy Scripts ...)?

File: test-scripts/tomcat-deployment/ansible/roles/tomcat/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+---
```
Comment (vidurananayakkara) [inline test-scripts/tomcat-deployment/ansible/roles/tomcat/tasks/main.yml]: Should this file have the `.yaml` extension [1] ?

[1] http://www.yaml.org/faq.html

File: test-scripts/tomcat-deployment/ansible/site.yml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+---
```
Comment (vidurananayakkara) [inline test-scripts/tomcat-deployment/ansible/site.yml]: Same as the above comment

File: test-scripts/tomcat-deployment/cf-templates/add-instances.yaml [hunk 4]
```diff
@@ -0,0 +1,73 @@
+# ----------------------------------------------------------------------------
+#
+# Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (vidurananayakkara) [inline test-scripts/tomcat-deployment/cf-templates/add-instances.yaml]: License year should be 2018

File: test-scripts/tomcat-deployment/deploy.sh [hunk 5]
```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bash
+# ----------------------------------------------------------------------------
+#
+# Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (vidurananayakkara) [inline test-scripts/tomcat-deployment/deploy.sh]: Same as the above comment

File: test-scripts/tomcat-deployment/infra-destroy.sh [hunk 6]
```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
+# ----------------------------------------------------------------------------
+#
+# Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (vidurananayakkara) [inline test-scripts/tomcat-deployment/infra-destroy.sh]: Same as the above comment

File: test-scripts/tomcat-deployment/infra-provision.sh [hunk 7]
```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bash
+# ----------------------------------------------------------------------------
+#
+# Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (vidurananayakkara) [inline test-scripts/tomcat-deployment/infra-provision.sh]: Same as the above comment

File: test-scripts/tomcat-deployment/infra-provision.sh [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+# Waiting till the AWS stack is created
+aws cloudformation wait stack-create-complete --stack-name $stackName
+
+
+# Write the public IP of the instance to a file
+# This will only work if you have a single output with the instance IP.
+aws cloudformation describe-stacks --stack-name $stackName | grep -o -P '(?<=OutputValue": ").*(?=")' > $outputFileName
+
+
```
Comment (vidurananayakkara) [inline test-scripts/tomcat-deployment/infra-provision.sh]: Unnecessary blank line (line 50)

File: test-scripts/tomcat-deployment/cf-templates/add-instances.yaml [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+          Value: wso2-tg-tomcat
+
+
+#####################################
+# Print details of the created stack
+#####################################
+Outputs:
+  serverHost:
+    Value: !Sub '${TGTomCatInstance.PublicIp}'
+    Description: DNS Name of the Instance
```
Comment (vidurananayakkara) [inline test-scripts/tomcat-deployment/cf-templates/add-instances.yaml]: Should this file end with a new line char? [1]

[1] https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

File: test-scripts/tomcat-deployment/infra-destroy.sh [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+
+echo "Infrastructure Destroy Initiated"
+
+# Deleting the created stack
+aws cloudformation delete-stack --stack-name $stackName
+
+# Deleting the created key pair
+aws ec2 delete-key-pair --key-name $keyPairName
+
+# TO-DO Hold the script till the instance is in terminated state
```
Comment (vidurananayakkara) [inline test-scripts/tomcat-deployment/infra-destroy.sh]: Should this file end with a new line char? [1]

[1] https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

File: test-scripts/tomcat-deployment/instance_map.csv [hunk 11]
```diff
@@ -0,0 +1,2 @@
+1,Ubuntu14.04|JDK1.7,22233445566
+2,RedHaT7,JDK1.8,1524875
```
Comment (vidurananayakkara) [inline test-scripts/tomcat-deployment/instance_map.csv]: Should this file end with a new line char? [1]

[1] https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

File: test-scripts/tomcat-deployment/run-scenarios.sh [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+# ----------------------------------------------------------------------------
+
+# This script is responsible for running the test scenarios
+
+# Populate the user.properties file
+
+cat deployment_eps > jmeter/user.properties
+
+cd jmeter
+jmeter -n -t 01-TestTomCat.jmx -p user.properties -l log.jtl
```
Comment (vidurananayakkara) [inline test-scripts/tomcat-deployment/run-scenarios.sh]: Should this file end with a new line char? [1]

[1] https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

File: test-scripts/tomcat-deployment/run-scenarios.sh [hunk 13]
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+# ----------------------------------------------------------------------------
+#
+# Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (vidurananayakkara) [inline test-scripts/tomcat-deployment/run-scenarios.sh]: License year should be 2018

File: test-scripts/tomcat-deployment/README.md (hunk 1 above)
Comment (yasassri) [reply test-scripts/tomcat-deployment/README.md]: This is Fixed with 90a076769115d759559419fa046eaf72e213abbb

File: test-scripts/tomcat-deployment/ansible/roles/tomcat/tasks/main.yml (hunk 2 above)
Comment (yasassri) [reply test-scripts/tomcat-deployment/ansible/roles/tomcat/tasks/main.yml]: Fixed with 0e68a5bf790f62747c93c792686566a1312d6575

File: test-scripts/tomcat-deployment/ansible/site.yml (hunk 3 above)
Comment (yasassri) [reply test-scripts/tomcat-deployment/ansible/site.yml]: Fixed with 0e68a5bf790f62747c93c792686566a1312d6575

File: test-scripts/tomcat-deployment/cf-templates/add-instances.yaml (hunk 4 above)
Comment (yasassri) [reply test-scripts/tomcat-deployment/cf-templates/add-instances.yaml]: Fixed with 1479a08a77d83ce3fe12f93a1f691490f7e8e860

File: test-scripts/tomcat-deployment/deploy.sh (hunk 5 above)
Comment (yasassri) [reply test-scripts/tomcat-deployment/deploy.sh]: Fixed with 1479a08a77d83ce3fe12f93a1f691490f7e8e860

File: test-scripts/tomcat-deployment/infra-destroy.sh (hunk 6 above)
Comment (yasassri) [reply test-scripts/tomcat-deployment/infra-destroy.sh]: Fixed with 1479a08a77d83ce3fe12f93a1f691490f7e8e860

File: test-scripts/tomcat-deployment/infra-provision.sh (hunk 7 above)
Comment (yasassri) [reply test-scripts/tomcat-deployment/infra-provision.sh]: Fixed with 1479a08a77d83ce3fe12f93a1f691490f7e8e860

File: test-scripts/tomcat-deployment/run-scenarios.sh (hunk 13 above)
Comment (yasassri) [reply test-scripts/tomcat-deployment/run-scenarios.sh]: Fixed with 1479a08a77d83ce3fe12f93a1f691490f7e8e860

File: test-scripts/tomcat-deployment/cf-templates/add-instances.yaml (hunk 9 above)
Comment (yasassri) [reply test-scripts/tomcat-deployment/cf-templates/add-instances.yaml]: This is fixed with : 3b681e445a2e8d0047be71d8f83145453662dc71

File: test-scripts/tomcat-deployment/infra-destroy.sh (hunk 10 above)
Comment (yasassri) [reply test-scripts/tomcat-deployment/infra-destroy.sh]: This is fixed with : 3b681e445a2e8d0047be71d8f83145453662dc71

File: test-scripts/tomcat-deployment/instance_map.csv (hunk 11 above)
Comment (yasassri) [reply test-scripts/tomcat-deployment/instance_map.csv]: This is fixed with : 3b681e445a2e8d0047be71d8f83145453662dc71

File: test-scripts/tomcat-deployment/run-scenarios.sh (hunk 12 above)
Comment (yasassri) [reply test-scripts/tomcat-deployment/run-scenarios.sh]: This is fixed with : 3b681e445a2e8d0047be71d8f83145453662dc71

File: test-scripts/tomcat-deployment/deploy.sh [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+
+# This script is responsible for deploying the Tomcat server on AWS instances
+
+# For the deployment script to run infra_eps and key.pem files should exist
+
+infraEPFileName=infra_eps
+outputFileName=deployment_eps
+
+#Extracting the IP of the tomcat server
+endpoint=$(cat $infraEPFileName)
```
Comment (kasunbg) [inline test-scripts/tomcat-deployment/deploy.sh]: This logic may fail when we add a new property to the infra provision output. We should rather parse the properties file and get the `serverHost` property.

This gist show how to parse properties files: https://gist.github.com/marcelbirkner/9b133f800d7d3fc5d828

File: test-scripts/tomcat-deployment/ansible/hosts [hunk 15]
```diff
@@ -0,0 +1,5 @@
+[tomcat-servers:vars]
+ansible_ssh_private_key_file=./key.pem
+
+[tomcat-servers]
+54.224.223.61
```
Comment (kasunbg) [inline test-scripts/tomcat-deployment/ansible/hosts]: It seems this file is re-generated by deploy.sh. So, shall we add a comment asking users to not modify this manually?

File: test-scripts/tomcat-deployment/deploy.sh [hunk 16]
```diff
@@ -0,0 +1,42 @@
...
+# under the License.
+#
+# ----------------------------------------------------------------------------
+
+# This script is responsible for deploying the Tomcat server on AWS instances
+
+# For the deployment script to run infra_eps and key.pem files should exist
+
+infraEPFileName=infra_eps
+outputFileName=deployment_eps
```
[the remark this answers is not in the document]
Comment (kasunbg) [reply test-scripts/tomcat-deployment/deploy.sh]: how about DEPLOYMENT_OUTPUT.properties name for the deployment output file?

File: test-scripts/tomcat-deployment/deploy.sh [hunk 17]
```diff
@@ -0,0 +1,42 @@
...
+
+#Extracting the IP of the tomcat server
+endpoint=$(cat $infraEPFileName)
+echo "The Tomcat Endpoint is set to : " $endpoint
+
+# Populate the ansible playbook host file
+echo -e "[tomcat-servers:vars]\nansible_ssh_private_key_file=./key.pem\n\n[tomcat-servers]\n$endpoint" > ansible/hosts
```
Comment (kasunbg) [inline test-scripts/tomcat-deployment/deploy.sh]: Can we simply replace the hosts entry?

File: test-scripts/tomcat-deployment/deploy.sh [hunk 18]
```diff
@@ -0,0 +1,42 @@
...
+
+# Executing ansible
+echo "Ansible execution started"
+export ANSIBLE_HOST_KEY_CHECKING=False
```
Comment (kasunbg) [inline test-scripts/tomcat-deployment/deploy.sh]: this should be at the very beginning.

File: test-scripts/tomcat-deployment/infra-destroy.sh [hunk 19]
```diff
@@ -0,0 +1,35 @@
...
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+# ----------------------------------------------------------------------------
+
+# This script is responsible for destroying the AWS instances and other AWS resources created
+
+stackName=TGdummy
+keyPairName=tgDummy
```
Comment (kasunbg) [inline test-scripts/tomcat-deployment/infra-destroy.sh]: .

File: test-scripts/tomcat-deployment/instance_map.csv (hunk 11 above)
Comment (kasunbg) [inline test-scripts/tomcat-deployment/instance_map.csv]: First line has a syntax error for the separate between second and third column!

File: test-scripts/tomcat-deployment/jmeter/01-TestTomCat.jmx [hunk 20]
```diff
@@ -0,0 +1,75 @@
...
+  <hashTree>
+    <TestPlan guiclass="TestPlanGui" testclass="TestPlan" testname="Test TomCat Server" enabled="true">
+      <stringProp name="TestPlan.comments"></stringProp>
+      <boolProp name="TestPlan.functional_mode">false</boolProp>
+      <boolProp name="TestPlan.serialize_threadgroups">false</boolProp>
+      <elementProp name="TestPlan.user_defined_variables" elementType="Arguments" guiclass="ArgumentsPanel" testclass="Arguments" testname="User Defined Variables" enabled="true">
+        <collectionProp name="Arguments.arguments">
+          <elementProp name="tomcat_host2" elementType="Argument">
+            <stringProp name="Argument.name">tomcat_host2</stringProp>
+            <stringProp name="Argument.value">http://35.153.166.223:8081</stringProp>
```
Comment (kasunbg) [inline test-scripts/tomcat-deployment/jmeter/01-TestTomCat.jmx]: hard-coded value.

File: test-scripts/tomcat-deployment/jmeter/user.properties [hunk 21]
```diff
@@ -0,0 +1 @@
+tomcat_host=http://54.224.223.61:8080
```
Comment (kasunbg) [inline test-scripts/tomcat-deployment/jmeter/user.properties]: Who modify this file? Shall we add DO_NOT_MODIFY; AUTO_GENERATED comment at the top? Is this really auto-generated?

File: test-scripts/tomcat-deployment/run-scenarios.sh [hunk 22]
```diff
@@ -0,0 +1,29 @@
...
+# ----------------------------------------------------------------------------
+
+# This script is responsible for running the test scenarios
+
+# Populate the user.properties file
+
+cat deployment_eps > jmeter/user.properties
+
+cd jmeter
+jmeter -n -t 01-TestTomCat.jmx -p user.properties -l log.jtl
```
Comment (kasunbg) [inline test-scripts/tomcat-deployment/run-scenarios.sh]: exit gracefully if jmeter is not found in the classpath.

**Pull request merged**